test/specs/core/mergeConfig.spec.js

```diff
@@ -69,4 +69,10 @@ describe('core::mergeConfig', function() {
     var merged = mergeConfig(defaults, { foo: 'bar' });
     expect(merged.foo).toEqual('bar');
   });
+
+  it('should allow setting custom default options', function() {
```

Comment:
Let me throw a bit of a twist in here :) [retry-axios](https://github.com/JustinBeckwith/retry-axios) supports adding a deep object to the request config like this:

```js
const myAxiosInstance = axios.create();
myAxiosInstance.defaults.raxConfig = {
  instance: myAxiosInstance,
  retry: 5
};
const interceptorId = rax.attach(myAxiosInstance);
const res = await myAxiosInstance.get('https://test.local');
```

It would be nice to add a test to cover for deep merging of customer properties that are objects as well.  Something like:

it('should allow merging custom objects in the config, function() {
  defaults.nesty = {
    nestedConfig: {
      propertyOnDefaultConfig: true
    }
  };
  var merged = mergeConfig(defaults, {
    nestedConfig: {
      propertyOnRequestConfig: true
    }
  });
  expect(merged.nestedConfig.propertyOnDefaultConfig).toEqual(true);
  expect(merged.nestedConfig.propertyOnRequestConfig).toEqual(true);
});
```